lib/matplotlib/backends/_backend_tk.py

```diff
@@ -328,26 +329,30 @@ def _get_key(self, event):
         # however this is not the case on "darwin", so double check that
         # we aren't adding repeat modifier flags to a modifier key.
         if sys.platform == 'win32':
-            modifiers = [(17, 'alt', 'alt'),
-                         (2, 'ctrl', 'control'),
+            modifiers = [(2, 'ctrl', 'control'),
+                         (17, 'alt', 'alt'),
+                         (0, 'shift', 'shift'),
                          ]
         elif sys.platform == 'darwin':
-            modifiers = [(3, 'super', 'super'),
+            modifiers = [(2, 'ctrl', 'control'),
                          (4, 'alt', 'alt'),
-                         (2, 'ctrl', 'control'),
+                         (0, 'shift', 'shift'),
+                         (3, 'super', 'super'),
                          ]
         else:
-            modifiers = [(6, 'super', 'super'),
+            modifiers = [(2, 'ctrl', 'control'),
                          (3, 'alt', 'alt'),
-                         (2, 'ctrl', 'control'),
+                         (0, 'shift', 'shift'),
+                         (6, 'super', 'super'),
                          ]
 
         if key is not None:
             # shift is not added to the keys as this is already accounted for
             for bitmask, prefix, key_name in modifiers:
                 if event.state & (1 << bitmask) and key_name not in key:
-                    key = '{0}+{1}'.format(prefix, key)
-
+                    if not (prefix == 'shift' and unikey):
+                        key = '{0}+{1}'.format(prefix, key)
+                    break
```

Comment:
Doesn't this break mean you can only add one modifier?

Reply:
If we hit ctrl + shift + alt, then we get ctrl, ctrl + shift and finally ctrl + alt. I noticed it, but haven't been able to work on this PR since I've been a little busy. If someone else want to continue with the work, go ahead! If not, I will work on this when I get more time. I'm sorry for the inconveniences.

Reply:
Assuming you meant ctrl+shift+alt for the last one, I think that is correct behavior, isn't it?